src/frequenz/sdk/actor/power_distributing/_component_pool_status_tracker.py

```diff
@@ -0,0 +1,163 @@
+            component_status_tracker_type: component status tracker to use
+                for tracking the status of the components.
+        """
+        self._component_ids = component_ids
+        self._max_data_age_sec = max_data_age_sec
+        self._max_blocking_duration_sec = max_blocking_duration_sec
+        self._component_status_sender = component_status_sender
+        self._component_status_tracker_type = component_status_tracker_type
+
+        # At first no component is working, we will get notification when they start
+        # working.
+        self._current_status = ComponentPoolStatus(working=set(), uncertain=set())
+
+        # Channel for sending results of requests to the components.
+        self._set_power_result_channel = Broadcast[SetPowerResult](
+            "component_request_status"
+        )
+        self._set_power_result_sender = self._set_power_result_channel.new_sender()
+        self._component_status_trackers: list[ComponentStatusTracker] = []
+        self._merged_status_receiver = self._make_merged_status_receiver()
+
+        self._task = asyncio.create_task(self._run())
+
+    async def join(self) -> None:
+        """Wait and return when the instance's task completes.
+
+        It will not terminate the instance, which can be done with the `stop` method.
+        """
+        await self._task
+
+    async def stop(self) -> None:
+        """Stop the ComponentPoolStatusTracker instance."""
+        await cancel_and_await(self._task)
+        await asyncio.gather(
+            *[tracker.stop() for tracker in self._component_status_trackers],
+        )
+        await self._merged_status_receiver.stop()
+
+    def _make_merged_status_receiver(
+        self,
+    ) -> Merge[ComponentStatus]:
+        status_receivers: list[Receiver[ComponentStatus]] = []
+
+        for component_id in self._component_ids:
+            channel: Broadcast[ComponentStatus] = Broadcast(
+                f"component_{component_id}_status"
+            )
+            tracker = self._component_status_tracker_type(
+                component_id=component_id,
+                max_data_age_sec=self._max_data_age_sec,
+                max_blocking_duration_sec=self._max_blocking_duration_sec,
+                status_sender=channel.new_sender(),
+                set_power_result_receiver=self._set_power_result_channel.new_receiver(),
+            )
+            self._component_status_trackers.append(tracker)
+            status_receivers.append(channel.new_receiver())
+        return Merge(*status_receivers)
+
+    async def _run(self) -> None:
+        """Start tracking component status."""
+        while True:
+            try:
+                await self._update_status()
+            except Exception as err:  # pylint: disable=broad-except
+                _logger.error(
+                    "ComponentPoolStatus failed with error: %s. Restarting.", err
+                )
```

Comment:
We might want to a short delay between restarts. This might be also out of scope for this PR but worth mentioning it

Reply:
Wow, I hadn't seen this.  I think this would never happen, because we don't close any of our channels and that's the only source of an exception here.

I think I'll just remove this exception handling, causing the actor to restart fully instead.

Reply:
Will do in the next PR.